backend/src/models/order.ts

```diff
@@ -0,0 +1,62 @@
+import mongoose from 'mongoose';
+
+const orderSchema = new mongoose.Schema({
+  restaurant: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Restaurant',
+  },
+  user: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+  },
+  deliveryDetails: {
+    email: {
+      type: String,
+      required: true,
+    },
+    name: {
+      type: String,
+      required: true,
+    },
+    addressLine1: {
+      type: String,
+      required: true,
+    },
+    city: {
+      type: String,
+      required: true,
+    },
+  },
+  cartItems: [
+    {
+      menuItemId: {
+        type: String,
+        required: true,
+      },
+      quantity: {
+        type: Number,
+        required: true,
+      },
+      name: {
+        type: String,
+        required: true,
+      },
+    },
+  ],
+  totalAmount: {
+    type: Number,
+  },
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/46#issuecomment-2423673837>#b6ffb4</a><span> - 10/19/2024, 08:13 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'totalAmount' field lacks validation to ensure it's a positive number.
</div>
<div id="fix">
<b>Fix:</b> Consider adding a 'min' validator to the 'totalAmount' field to ensure it's always a positive number, and make it a required field for data integrity.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -46,6 +46,8 @@
    totalAmount: {
      type: Number,
 +    min: [0, 'Total amount cannot be negative'],
 +    required: true,
    },
    status: {
      type: String,
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged